addons/osfstorage/models.py

```diff
@@ -342,6 +352,14 @@ def is_checked_out(self):
                 pass
         return False
 
+    @property
+    def is_preprint_primary(self):
```

Comment:
A recursive CTE would work, but I don't know of any Django magic for this. Check out `AbstractNodeQuerySet.get_children` if you're feeling adventurous -- you'd want something similar.